File: drawee-backends/drawee-pipeline/src/main/java/com/facebook/drawee/backends/pipeline/info/VisibilityState.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.drawee.backends.pipeline.info;

import static com.facebook.drawee.backends.pipeline.info.VisibilityState.INVISIBLE;
import static com.facebook.drawee.backends.pipeline.info.VisibilityState.VISIBLE;
import static java.lang.annotation.RetentionPolicy.SOURCE;

import androidx.annotation.IntDef;
import java.lang.annotation.Retention;

@Retention(SOURCE)
@IntDef({
  VISIBLE, INVISIBLE,
})
public @interface VisibilityState {

  int UNKNOWN = -1;
  int VISIBLE = 1;
  int INVISIBLE = 2;
}

File: ui-common/src/main/java/com/facebook/fresco/ui/common/VisibilityState.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.ui.common

enum class VisibilityState(val value: Int) {
  UNKNOWN(-1),
  VISIBLE(1),
  INVISIBLE(2);

  companion object {
    private val VALUES = values()

    fun fromInt(value: Int): VisibilityState? = VALUES.firstOrNull { it.value == value }
  }
}
